From ce1e276a657038a3558ce2cd75558120af8fd605 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Thu, 04 Jan 2024 13:40:42 +0200
Subject: [PATCH] [SW-169995] efuse: do not return error if initialized

if efuse already has been initialized it is not an error condition
and there is no need to return error.

Change-Id: I0e7368dfe25f7c347df298f061e2ef4129009fed
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/lib/common/efuse.c b/lib/common/efuse.c
index 365731f..fd13f75 100644
--- a/lib/common/efuse.c
+++ b/lib/common/efuse.c
@@ -50,7 +50,7 @@
 	u32 read_reg;
 
 	if (efuse_init == 1)
-		return -1;
+		return 0;
 
 	regmap_writel(efuse.addr_range, efuse_base + efuse.read_address_off);
 	regmap_writel(1, efuse_base + efuse.read_ctrl_off);
